pokedex-with-bit/executors.py

```diff
@@ -0,0 +1,271 @@
+            import tensorflow as tf
+            cpus = tf.config.experimental.list_physical_devices(
+                device_type='CPU')
+            gpus = tf.config.experimental.list_physical_devices(
+                device_type='GPU')
+            if self.on_gpu and len(gpus) > 0:
+                cpus.append(gpus[0])
+            tf.config.experimental.set_visible_devices(devices=cpus)
+            self.logger.info(f'model_path: {self.model_path}')
+            #_model = tf.saved_model.load(self.model_path)
+            from tensorflow.python.keras.models import load_model
+            _model = load_model(self.model_path)
+            self.model = _model.signatures['serving_default']
+            self._get_input = tf.convert_to_tensor
+        else:
+            raise PretrainedModelFileDoesNotExist(f'model at {self.model_path} does not exist')
+
+    @requests
+    def encode(self, docs: DocumentArray, **kwargs) -> DocumentArray:
+        """
+        Encode data into a ndarray of `B x D`.
+        Where `B` is the batch size and `D` is the Dimension.
+
+        :param docs: an array in size `B`
+        :return: an ndarray in size `B x D`.
+        """
+        data = np.zeros((docs.__len__,) + docs[0].blob.shape)
+        for index, doc in enumerate(docs):
+            data[index] = doc.blob
+        if self.channel_axis != -1:
+            data = np.moveaxis(data, self.channel_axis, -1)
+        _output = self.model(self._get_input(data.astype(np.float32)))
+        output = _output['output_1'].numpy()
+        for index, doc in enumerate(docs):
+            doc.embedding = output[index]
+        return docs
+
+
+class DocVectorIndexer(Executor):
+    def __init__(self, index_file_name: str, **kwargs):
+        super().__init__(**kwargs)
+        self.index_file_name = index_file_name
+        if os.path.exists(self.save_path):
+            self._docs = DocumentArray.load(self.save_path)
+        else:
+            self._docs = DocumentArray()
+
+    @property
+    def save_path(self):
+        if not os.path.exists(self.workspace):
+            os.makedirs(self.workspace)
+        return os.path.join(self.workspace, self.index_file_name)
+
+    def close(self):
+        self._docs.save(self.save_path)
+
+    @requests(on='/index')
+    def index(self, docs: 'DocumentArray', **kwargs):
+        self._docs.extend(docs)
+
+    @requests(on='/search')
+    def search(self, docs: 'DocumentArray', parameters: Dict, **kwargs):
+        a = np.stack(docs.get_attributes('embedding'))
+        b = np.stack(self._docs.get_attributes('embedding'))
+        q_emb = _ext_A(_norm(a))
+        d_emb = _ext_B(_norm(b))
+        dists = _cosine(q_emb, d_emb)
+        idx, dist = self._get_sorted_top_k(dists, int(parameters['top_k']))
+        for _q, _ids, _dists in zip(docs, idx, dist):
+            for _id, _dist in zip(_ids, _dists):
+                d = Document(self._docs[int(_id)], copy=True)
```

Comment:
Could you change this `d` to `doc`? We want to try all examples have same variable names :)

Reply:
Done in https://github.com/jina-ai/examples/pull/626/commits/5d1abcf1e00c2f122b41d15bfdc65941d085711e